### PR TITLE
Simplify the SDK story

### PR DESCRIPTION
**Pull request by ianthehat**
This makes it so that `@go_sdk` is always the active sdk.
This means that the version is now set by which sdk you map to that name, not by
constraints in the toolchain. It also means that you can find the sdk and source
by a common name without going through the active toolchain.

This is a prerequiste for compiling the stdlib on demand (see #814)

### CODE REVIEW

File: go/toolchains.rst [hunk 1]
```diff
@@ -244,47 +249,88 @@ SDK will be used.
...
+|                                                                                                  |
++--------------------------------+-----------------------------+-----------------------------------+
+
+
+go_host_sdk
+~~~~~~~~~~~
+
+This detects the host Go SDK for use in toolchains.
+
+It first checks the GO_ROOT and then searches the PATH. You can achive the same result by setting
```
Comment (jayconrod) [inline go/toolchains.rst]: `GOROOT`

File: go/toolchains.rst [hunk 2]
```diff
@@ -36,17 +36,22 @@ The SDK
...
+we hope to make that an on demand step in the future.
 
 SDKs are specific to the host they are running on and the version of Go they want to use
 but not the target they compile for. The Go SDK is naturally `cross compiling`_.
 
-The Go rules already adds a go_sdk_ rule for all the host and Go version pairs that are shipped
-on the main Go language website, so there should be no need to declare one of these unless you
-need a `forked version of Go`_\, however you may want to `control the version`_ or use the
-`installed sdk`_.
+If you don't do anything special, the go rules will already add a go_sdk for you, defaulting to the
```
Comment (jayconrod) [inline go/toolchains.rst]: If you don't do anything special, the Go rules will download the most recent official SDK for your host.

File: go/toolchain/toolchains.bzl [hunk 3]
```diff
@@ -1,124 +1,149 @@
...
-      struct(
-          name = "1.7.6",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"]
-      ),
-      struct(
-          name = "1.7.5",
-          hosts = ["darwin_amd64", "linux_amd64"],
-      ),
-  ]
+HOSTS = (
```
Comment (jayconrod) [inline go/toolchain/toolchains.bzl]: Consider renaming to `HOST_TARGETS` or something that expresses that this is a map from host platforms to target platforms.

File: go/private/toolchain.bzl [hunk 4]
```diff
@@ -20,11 +20,46 @@ def executable_extension(ctx):
...
   _sdk_build_file(ctx, path)
+  _local_sdk(ctx, path)
+  _prepare(ctx)
 
 go_host_sdk = repository_rule(_go_host_sdk_impl, environ = ["GOROOT"])
-"""go_host_sdk is a specialization of go_sdk just to add the GOROOT dependancy"""
+
+def _go_download_sdk_impl(ctx):
+  if ctx.os.name == 'linux':
+      host = "linux_amd64"
```
Comment (jayconrod) [inline go/private/toolchain.bzl]: Nit: 2-space indent below `if`

File: go/private/toolchain.bzl [hunk 5]
```diff
@@ -37,20 +72,24 @@ def _go_sdk_impl(ctx):
...
   if "TMP" in ctx.os.environ:
     tmp = ctx.os.environ["TMP"]
     ctx.symlink(tmp, "tmp")
   else:
     ctx.file("tmp/ignore", content="") # make a file to force the directory to exist
     tmp = str(ctx.path("tmp").realpath)
-
   # Build the standard library for valid cross compile platforms
   #TODO: fix standard library cross compilation
   if ctx.name.endswith("linux_amd64") and ctx.os.name == "linux":
```
Comment (ixdy) [inline go/private/toolchain.bzl]: I think this PR broke this section - `ctx.name` always seems to be `go_sdk` now, so none of these conditionals match.

**Pull request merged**